pkg/dogstatsd/parse_metrics.go

```diff
@@ -0,0 +1,142 @@
+package dogstatsd
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+)
+
+type metricType int
+
+const (
+	gaugeType metricType = iota
+	countType
+	distributionType
+	histogramType
+	setType
+	timingType
+)
+
+var (
+	gaugeSymbol        = []byte("g")
+	countSymbol        = []byte("c")
+	histogramSymbol    = []byte("h")
+	distributionSymbol = []byte("d")
+	setSymbol          = []byte("s")
+	timingSymbol       = []byte("ms")
+
+	tagsFieldPrefix       = []byte("#")
+	sampleRateFieldPrefix = []byte("@")
+)
+
+type dogstatsdMetricSample struct {
```

Comment:
An easy win here would be to have this instead:
```
type dogstatsdMetricSample struct {
	value      float64 // 8 bytes
	sampleRate float64 // 8 bytes
	name       []byte // 12 bytes
	setValue   []byte // 12 bytes
	tags       [][]byte // 12 bytes
	metricType metricType // 4 bytes
}
```

This way, because Go is allocating consecutive packets of 8 bytes, we would have a struct of 56 bytes instead of 64 bytes. May not true on all arch but it's an easy win for the most used proc architecture :)
I've checked other structs and seeing their content types, I don't think this can be done elsewhere.

Reply:
Discussed this on chat but this is already optimal on 64bit